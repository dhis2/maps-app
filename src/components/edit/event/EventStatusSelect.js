import React from 'react';
import PropTypes from 'prop-types';
import i18n from '@dhis2/d2-i18n';
import { SelectField } from '../../core';
import {
    getEventStatuses,
    EVENT_STATUS_ALL,
} from '../../../constants/eventStatuses';

const EventStatusSelect = ({
    value = EVENT_STATUS_ALL,
    onChange,
    className,
}) => (
    <SelectField
        label={i18n.t('Event status')}
        items={getEventStatuses()}
        value={value}
        onChange={valueType => onChange(valueType.id)}
        className={className}
    />
);

EventStatusSelect.propTypes = {
    value: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    className: PropTypes.string,
};

export default EventStatusSelect;
