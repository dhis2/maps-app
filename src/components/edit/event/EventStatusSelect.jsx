import i18n from '@dhis2/d2-i18n'
import PropTypes from 'prop-types'
import React from 'react'
import {
    getEventStatuses,
    EVENT_STATUS_ALL,
} from '../../../constants/eventStatuses.js'
import { SelectField } from '../../core/index.js'

const EventStatusSelect = ({
    value = EVENT_STATUS_ALL,
    onChange,
    className,
}) => (
    <SelectField
        label={i18n.t('Event status')}
        items={getEventStatuses()}
        value={value}
        onChange={(valueType) => onChange(valueType.id)}
        className={className}
    />
)

EventStatusSelect.propTypes = {
    onChange: PropTypes.func.isRequired,
    className: PropTypes.string,
    value: PropTypes.string,
}

export default EventStatusSelect
