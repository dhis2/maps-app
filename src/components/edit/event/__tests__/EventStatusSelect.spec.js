import React from 'react';
import { render } from '@testing-library/react';
import EventStatusSelect from '../EventStatusSelect';

test('EventStatusSelect displays a Select element', () => {
    const { container } = render(
        <EventStatusSelect onChange={jest.fn()} className="testClass" />
    );
    expect(container).toMatchSnapshot();
});
