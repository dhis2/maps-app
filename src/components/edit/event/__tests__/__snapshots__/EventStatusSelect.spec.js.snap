// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventStatusSelect displays a Select element 1`] = `
<div>
  <div
    class="inputField testClass"
  >
    <div
      class="jsx-1141087211 "
      data-test="dhis2-uiwidgets-singleselectfield"
    >
      <label
        class="jsx-2718078600 "
        data-test="dhis2-uiwidgets-singleselectfield-label"
      >
        <span
          class="jsx-2718078600"
        >
          Event status
        </span>
      </label>
      <div
        class="jsx-3111771310 "
        data-test="dhis2-uiwidgets-singleselectfield-content"
      >
        <div
          class="jsx-3415779570 "
          data-test="dhis2-uicore-box"
        >
          <div
            class="jsx-2510973279 root"
            data-test="dhis2-uicore-singleselect"
          >
            <div
              class="jsx-2510973279 root-input"
            >
              <div
                data-test="dhis2-uicore-select"
              >
                <div
                  class="jsx-297747822 root dense"
                  data-test="dhis2-uicore-select-input"
                  tabindex="0"
                >
                  <div
                    class="jsx-297747822 root-children"
                  >
                    <div
                      class="jsx-2240447017 jsx-4123978364 root"
                    >
                      <div
                        class="jsx-2240447017 jsx-4123978364 root-input"
                      >
                        <div
                          class="jsx-3691749911 root"
                        >
                          All
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="jsx-297747822 root-right"
                  >
                    <svg
                      class="jsx-595155045 "
                      version="1.1"
                      viewBox="0 0 16 16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        class="jsx-595155045"
                        fill="none"
                        fill-rule="evenodd"
                        id="icon/16/arrow-down"
                        stroke="none"
                        stroke-width="1"
                      >
                        <path
                          class="jsx-595155045"
                          d="M7.29289,11.7071 C7.68342,12.0976 8.31658,12.0976 8.70711,11.7071 L13.7071,6.70711 C14.0976,6.31658 14.0976,5.68342 13.7071,5.29289 C13.3166,4.90237 12.6834,4.90237 12.2929,5.29289 L8,9.58579 L3.70711,5.29289 C3.31658,4.90237 2.68342,4.90237 2.29289,5.29289 C1.90237,5.68342 1.90237,6.31658 2.29289,6.70711 L7.29289,11.7071 Z"
                          fill="#4A5768"
                          id="arrow-down"
                        />
                      </g>
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
